compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5233,14 +5231,11 @@ static TR::Register *inlineSimpleAtomicUpdate(TR::Node *node, bool isAddOp, bool
       newValueReg = cg->allocateRegister();
 
       if (isDeltaImmediateShifted)
-         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node,
-               newValueReg, oldValueReg, ((delta & 0xFFFF0000) >> 16));
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, newValueReg, oldValueReg, delta >> 16);
       else if (isDeltaImmediate)
-         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node,
-               newValueReg, oldValueReg, delta);
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, newValueReg, oldValueReg, delta);
       else
-         generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, newValueReg,
-               oldValueReg, deltaReg);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, newValueReg, oldValueReg, deltaReg);
       }
    else
       {
```

Comment:
The outer "if" condition seemed broken already.  The intention is to test the value fits in 32bit, but 
                        && deltaChild->getLongInt() <= UPPER_IMMED
                        && deltaChild->getLongInt() >= LOWER_IMMED
means it fits within 16-bit. Such that, masking any bugs maybe.

Reply:
I see the problem there. I'll change that to check against `INT32_MAX` and `INT32_MIN` instead.

Reply:
xlC didn't seem to like the use of `INT32_MAX` and `INT32_MIN`. As such, I replaced them with 64 bit constants instead.